Exercise_2/main.cpp

```diff
@@ -0,0 +1,179 @@
+#include <string>
+#include <locale>
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+#include <vector>
+#include <limits>
+#include <climits>
+#include <list>
+
+using namespace std;
+
+struct nodes {
+    int start_node;
+    int end_node;
+    int distance;
+};
+
+struct table {
+    int vertex;
+    int distance;
+    int previos_vector;
+};
+
+
+vector<int> create_set_of_values_from_file(const vector<nodes> &all_data){
+    vector<int> set_of_values = {};
+    for(int i=0; i < all_data.size(); ++i){
+        if(find(set_of_values.begin(), set_of_values.end(), all_data[i].start_node) == set_of_values.end()){
+            set_of_values.push_back(all_data[i].start_node);
+        }
+    }
+    return set_of_values;
+}
+
+vector<table> create_table_from_file(const vector<int> &set_of_values, int start_node){
+    vector<table> result_table = {};
+    table row = {};
+    for (int i=0;i <set_of_values.size(); ++i) {
```

Comment:
A range-based for loop is preferred:
`for(const auto value : set_of_values) ...`